src/namicorn.ts

```diff
@@ -84,6 +84,32 @@ class Namicorn {
     }
   }
 
+  /**
+   * Resolves the ipfs hash configured for domain records on ZNS
+   * @param domain - domain name
+   * @throws ResolutionError
+   * - UnsupportedDomain if it is not supported by ZNS
+   * @returns a Promise that resolves in ipfsHash
+   */
+  async ipfsHash(domain: string): Promise<string> {
+    if (!this.zns.isSupportedDomain(domain))
```

Comment:
lets have less of a hardcode here by having `NamingService#supportsRecords(): boolean` and use it like:

```
const method = this.getNamingService(domain);
if (!method.supportsRecords) throw new ...;
return await method.ipfsHash();
```